config/configcompression/compressionType.go

```diff
@@ -0,0 +1,49 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package configcompression // import "go.opentelemetry.io/collector/config/configcompression"
+
+import "fmt"
+
+type CompressionType string
+
+const (
+	CompressionGzip    CompressionType = "gzip"
+	CompressionZlib    CompressionType = "zlib"
+	CompressionDeflate CompressionType = "deflate"
+	CompressionSnappy  CompressionType = "snappy"
+	CompressionZstd    CompressionType = "zstd"
+	CompressionNone    CompressionType = "none"
+	CompressionEmpty   CompressionType = ""
```

Comment:
Interested about the "empty" value... Should it be a public value for it? What is the use-case for it?

Reply:
`CompressionEmpty` and `CompressionNone` was used in a different package before but now they are only used in here. I'll make them unexported.